vscode/open/command.py

```diff
@@ -171,74 +202,56 @@ def command(shell, args: list):
             version=project.distro,
         )
 
-        DEVCONTAINER_TEMPLATE["image"] = image
+        container_configuration.image = image
+
+        # If devcontainer_configuration.dockerComposeFile is not specified, use the default one
+        if devcontainer_configuration.dockerComposeFile is None:
+            devcontainer_configuration.dockerComposeFile = "docker-compose.yml"
+
+        # Add workspace folder if not specified in the devcontainer_configuration
+        if devcontainer_configuration.workspaceFolder is None:
+            devcontainer_configuration.workspaceFolder = destination_src
+
+        # Create docker-compose and devcontainer.json files in .devcontainer folder
+        if not os.path.exists('.devcontainer') and not os.path.isdir('.devcontainer'):
+            os.mkdir('.devcontainer')
 
-        # docker arguments
-        if not parsed.docker_args:
-            parsed.docker_args = []
+        compose_file = COMPOSE_FILE_TEMPLATE
 
-        # escape spaces in arguments
-        parsed.docker_args = [a.replace(" ", "\\ ") for a in parsed.docker_args]
+        compose_file["services"]={devcontainer_configuration.container : container_configuration.__dict__}
```

Comment:
no need to use `devcontainer_configuration.container` here. We can call the service always `devcontainer` regardless of the actual configuration, it is just a name. Also, better for when you do `docker ps`, you will see `devcontainer` in the name of the container.

Reply:
Ok. We can change this in the future if we want to be able to specify a specific `service` to use as our devcontainer.